model/dependencies.go

```diff
@@ -58,6 +58,14 @@ func (di *dependencyIncluder) include(initialDeps []TVPair, activationInfo *spec
 	for pair, addToActivationInfo := range di.deactivateGeneratedDeps {
 		if addToActivationInfo {
 			activationInfo.activationTasks[pair.Variant] = append(activationInfo.activationTasks[pair.Variant], pair.TaskName)
+			bvt := di.Project.FindTaskForVariant(pair.TaskName, pair.Variant)
+			// Also need to deactivate the dependencies of the newly deactivated tasks
+			if bvt != nil {
+				deps := di.expandDependencies(pair, bvt.DependsOn)
```

Comment:
Wouldn't we want to handle this using the deactivation map instead? I might be thinking about this wrong since it's getting complex but, if taskA depends on taskB which depends on taskB2 (in your example), what if taskB2 _is_ supposed to be activated because something else does depend on it?

Reply:
Introduced a new `isRoot` parameter to the `handle` function which does the work of updating the deactivation map (if `isRoot` is true).

This should work because although `handle` computes dependencies recursively based off an [initialDeps](https://github.com/evergreen-ci/evergreen/blob/main/model/dependencies.go#L44) list provided by the generate tasks function, the only tasks that could have their activation initially set are the initialDeps, so we only need to propagate the `deactivateGeneratedDeps` at the root level of each recursion, as long as we update the map for the full recursive list of dependencies that need to exist to support the initial dependency.